src/main/java/net/fabricmc/loader/impl/game/minecraft/Log4jLogHandler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.impl.game.minecraft;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.fabricmc.loader.util.log.Log;
+import net.fabricmc.loader.util.log.LogCategory;
+import net.fabricmc.loader.util.log.LogHandler;
+import net.fabricmc.loader.util.log.LogLevel;
+
+final class Log4jLogHandler implements LogHandler {
```

Comment:
Might be worth putting a note somewhere saying this should not be used by mods and log4j should be used instead?

Reply:
As discussed privately this will be addressed by some general mechanism that discourages access to impl packages.